### PR TITLE
work on entities and enchantments

### PR DESCRIPTION
**Pull request by eristhea**
brings n/m/entity to 100%!

### CODE REVIEW

**Comment by supersaiyansubtlety** [comment]
you removed wip but it's still a draft
edit: maybe intentional until #598 is merged?

File: mappings/net/minecraft/entity/Leashable.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/entity/Leashable.mapping]:
```suggestion
	METHOD m_feakufat hasLeashData ()Z
```
more direct, and `mayBeLeashed()` sounds like a duplicate of `canBeLeashed`

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]
just one confusing name

File: mappings/net/minecraft/entity/Entity.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
	METHOD m_gykcjypz addPortalTicket (Lnet/minecraft/unmapped/C_hynzadkk;)V
```
one more
`place` seems like the wrong verb for a ticket, and the fact that it's tied to a chunk seems like an implementation detail

**Comment by supersaiyansubtlety** [comment]
these conflict with #604
```
net/minecraft/component/type/ItemEnchantmentsComponent.mapping
net/minecraft/enchantment/Enchantment.mapping
net/minecraft/enchantment/EnchantmentLevelEntry.mapping
net/minecraft/enchantment/Enchantments.mapping
```

**Comment by eristhea** [comment]
> you removed wip but it's still a draft edit: maybe intentional until #598 is merged?

yes, draft until the FCP prs are merged

File: mappings/net/minecraft/entity/Entity.mapping
Comment (eristhea) [reply mappings/net/minecraft/entity/Entity.mapping]: got it, i'll update both places

**Review by supersaiyansubtlety — APPROVED** [review]

File: mappings/net/minecraft/util/math/Box.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/util/math/Box.mapping]:
```suggestion
		COMMENT Creates a box that is translated by {@code vec.x}, {@code vec.y},
		COMMENT {@code vec.z} on each axis from this box.
```

**Review by supersaiyansubtlety — APPROVED** [review]

**Comment by eristhea** [comment]
sorry for torturing you with my wip system that is known to only myself

File: mappings/net/minecraft/entity/mob/MobEntity.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/entity/mob/MobEntity.mapping]:
```suggestion
	METHOD m_cyqouyby dropPreservedEquipment ()V
```
it doesn't drop all equipment because `m_mapkswnb` only drops equipment in slots that are guaranteed to drop

File: mappings/net/minecraft/entity/mob/MobEntity.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/entity/mob/MobEntity.mapping]:
```suggestion
	METHOD m_mapkswnb dropPreservedEquipment (Ljava/util/function/Predicate;)Ljava/util/Set;
		COMMENT Drop equipment from slots that are guaranteed to drop and which match the passed {@code shouldDrop} predicate
		COMMENT @return slots containing equipment that doesn't match the passed {@code shouldDrop} predicate
```
moving this here as I solve #607's conflicts with this PR

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]
#607's going to have to wait for this PR to merge

File: mappings/net/minecraft/entity/mob/MobEntity.mapping
Comment (eristhea) [reply mappings/net/minecraft/entity/mob/MobEntity.mapping]: ah ok, i missed that
good catch!

**Review by supersaiyansubtlety — APPROVED** [review]

**Comment by eristhea** [comment]
FCP for three days due to inactivity.

File: mappings/net/minecraft/entity/LivingEntity.mapping
Comment (IotaBread) [inline mappings/net/minecraft/entity/LivingEntity.mapping:484]: What is this?

**Review by IotaBread — APPROVED** [review]

File: mappings/net/minecraft/entity/LivingEntity.mapping
Comment (eristhea) [reply mappings/net/minecraft/entity/LivingEntity.mapping:484]: the third parameter? a boolean, whether the kill was by a player

**Pull request merged**
